logical/framework/field_type.go

```diff
@@ -13,6 +13,16 @@ const (
 	// TypeDurationSecond represent as seconds, this can be either an
 	// integer or go duration format string (e.g. 24h)
 	TypeDurationSecond
+
+	// TypeSlice represents a slice of any type
+	TypeSlice
+	// TypeStringSlice is a helper for TypeSlice that returns a sanitized
```

Comment:
Hrm. I think trimming, sure. I'd leave deduping and/or sorting to the backend, we may not be thinking of totally rational use cases for having duplicates or specific orders.